JoinColony/colonyDapp
# Add Stakes tab to Token Activation Popover

## Description

This PR adds the a 'Stakes' tab into the token activation popover and provides a query of claimable `motionId`'s that can use used by the 'Claim All' stakes button and the staked motions list.

![stakes-tab](https://user-images.githubusercontent.com/33682027/151794794-dfb875c1-0c8b-413f-8157-189e966a7a80.png)

NOTE: I went around in circles on this with creating a custom query, using existing queries, using resolvers... I ended up just using the existing `useSubgraphUserMotionTokenEventsQuery` query and using a filter for simplicity. I would love to hear any suggestions for alternative / better ways to get the list of claimable motions.

Or, any thoughts on handling requests greater then 1,000 (given the query only gets staked events and won't remove them from the query if claimed). Although, I figure this is not important and by sorting query date by descending we can capture majority of cases (unless they are staking a ridiculous amount and never claiming).

**New stuff** ✨

* Added/Re-enabled ClaimsTab in `TokenActivationContent.tsx`
* Added `walletAddress` to UserTokenActivationButton and  TokenActivationContent
* Added query for claimable motions
* Added resolver `claimableStakedMotions` and query `ClaimableStakedMotions`

## Testing
* You will need to have the Motions & Disputes extension installed.
* Create some motions and finalize, but don't claim (if you do a lot, try claiming some). 
Reminder, you can speed up the chain 3 days using command: `curl -H "Content-Type: application/json" -X POST --data '{"jsonrpc":"2.0","method":"evm_increaseTime","params":[260000],"id": 1}' localhost:8545 && curl -H "Content-Type: application/json" -X POST --data '{"jsonrpc":"2.0","method":"evm_mine","params":[]}' localhost:8545`
* Navigate to the `TokenActivation` popover and click on the 'Stakes' tab.
* You should see a loader while waiting for the query, then the relevant messages 'There are no stakes to claim.' or '[Claimable stakes appear here]' (this is just a temporary message).
* Console log `unclaimedMotions` to see array of claimable motionIds.

Resolves #3152
